common/test/acceptance/tests/lms/test_lms.py

```diff
@@ -680,3 +681,73 @@ def test_python_execution_in_problem(self):
         problem_page.fill_answer("4")
         problem_page.click_check()
         self.assertFalse(problem_page.is_correct())
+
+
+class EntranceExamTest(UniqueCourseTest):
+    """
+    Tests that course has an entrance exam.
+    """
+
+    def setUp(self):
+        """
+        Initialize pages and install a course fixture.
+        """
+        super(EntranceExamTest, self).setUp()
+
+        CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        ).install()
+
+        self.course_info_page = CourseInfoPage(self.browser, self.course_id)
+        self.settings_page = SettingsPage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+        # Auto-auth register for the course
+        AutoAuthPage(self.browser, course_id=self.course_id).visit()
+
+    def test_entrance_exam_section(self):
```

Comment:
I think you should have several Bok Choy tests for the different scenarios:

1. As a student, I can't enter the course before completing the entrance exam
2. As a student, the course becomes available to me as soon as I complete the entrance exam
3. As a student, I am shown an error message if my entrance exam score is too low

I'm sure there are others, but those are the ones that occur to me right now.